### DISCUSSION TITLE
Business rules dependent on property calculated by another BR do not evaluate.

### DISCUSSION THREAD

**Discussion by bujma** [MarimerLLC/csla#4214] [Questions] [answered]
Hello.

We are using CSLA in our project. We have a few business objects that work, but in each we have problems when it comes to dependency of calculated properties: calculations that depend on another calculated property do not fire.

Let's say we have this simple example:
* A * (B + C) = R1
* there is another calculation returning string result R2 according to R1 value
* for a reason, result of (B + C) is important information itself, not just internal part of final R1 calculation. Therefor its calculation is a separate rule.

Findings:
* (B + C) calculation is fired after any change of B or C
* neither R1 nor R2 calculations get started after change of B or C
* calculation R1 starts only after change of A.

Where is the problem?

Details are in attached solution ([DependentRulesProblem.zip](https://github.com/user-attachments/files/16932394/DependentRulesProblem.zip)).  
In the code: A = `InputValue1`, B = `InputValue2Part1`, C = `InputValue2Part2`, (B + C) = `CalculatedValue2`, R1 = `Result1Number`, R2 = `Result2String`.

Thanks much.

**Comment by hurcane** [comment, marked as answer]
The default behavior of business rules is to **not** cascade property changes. This is a safeguard to avoid unexpected infinite loops.

To enable the desired behavior, add this line to the class constructor:
```BusinessRules.CascadeOnDirtyProperties = true;```

**Reply by bujma** [reply]
That was super fast and even works 😉. How could I overlook that property... thank you very much.

**Reply by ossendorf-at-hoelscher** [reply]
I think the constructor is too early. Because the `BusinessRules` depend on the `ApplicationContext` which is not set in the constructor.
I'm setting this property in the `OnApplicationContextSet` method which you can override.

**Reply by bujma** [reply]
Just a note for my case: I did not mention that for current moment we have to use old version 5.5.2, where `OnApplicationContextSet` is not present and we live in "old" CSLA environment.  
Thanks for your note, I will use this info in the future when we migrate to the latest version. 🤝
